app/controllers/works_controller.rb

```diff
@@ -91,4 +103,12 @@ def category_from_work
     render_404 unless @work
     @media_category = @work.category.downcase.pluralize
   end
+
+  def require_login
+    unless session[:user_id]
+
+      flash[:error] = "You must be logged in to do that."
```

Comment:
I like that this is a controller filter, but you've got much the same code both here and in the `users` controller. Could you DRY it up further by putting this in the application controller?